### PR TITLE
[REFACTOR] Refactoring database structure

### PR DESCRIPTION
**Pull request by M4ST3RX**
#### Issue or feature explanation
Before there was lots of if statements to determine which database connection to use. Now it is changed to use class functions which makes it easier to add more database type support. Also added language to error file.

#### Proposed solution/change
- Created custom connection classes with the same function names but different content.
- When connecting to the database dynamically figure out the class which extends `Illuminate\Database\Connection` and call the different functions. See commits.

### CODE REVIEW

File: resources/assets/js/components/MainContent/BrowseMode/Table/Table.vue [hunk 1]
```diff
@@ -103,6 +103,9 @@
       },
 
       resetFocus: function(ev) {
+      	if(this.view.cell.editing){
```
Comment (QuintenJustus) [inline resources/assets/js/components/MainContent/BrowseMode/Table/Table.vue]: Move to a different feature branch and create a new pull request please. This is a separate feature.

File: src/Classes/Database/MysqlConnection.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    public function __construct()
+    {
+        parent::__construct($this->getPdo());
+        $this->connection = new Connection($this->getPdo());
+    }
+
+    /**
+     * @return MysqlConnection
+     */
+    public function getConnection(): MysqlConnection
```
Comment (QuintenJustus) [inline src/Classes/Database/MysqlConnection.php]: The `: MysqlConnection ` syntax should not be used yet as PHP 7.4 isn't that widely used at the moment. Try and keep it atleast 7.1 compatible.

File: src/Classes/Database/MysqlConnection.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Protoqol\Prequel\Classes\Database;
+
+
+use Illuminate\Database\Connection;
+use Illuminate\Database\Query\Processors\MySqlProcessor;
+use Illuminate\Database\Query\Grammars\MySqlGrammar;
+
+class MysqlConnection extends Connection
```
Comment (QuintenJustus) [inline src/Classes/Database/MysqlConnection.php]: Use [Pascal case](http://wiki.c2.com/?PascalCase) for classes, so: `MySqlConnection`.

File: tests/Unit/Classes/Database/DatabaseTest.php [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Protoqol\Prequel\Tests\Unit\Classes\Database;
+
+use Protoqol\Prequel\Tests\TestCase;
+
+class DatabaseTest extends TestCase
```
Comment (QuintenJustus) [inline tests/Unit/Classes/Database/DatabaseTest.php]: See the Travis CI results.

**Review by QuintenJustus — DISMISSED** [review]
See comments

File: resources/assets/js/components/MainContent/BrowseMode/Table/Table.vue [hunk 5]
```diff
@@ -114,7 +117,8 @@
        *
        * @TODO
        */
-      saveModifiedData: function() {
+      saveModifiedData: function(el) {
+      	console.log(el)
```
Comment (James-N-M) [inline resources/assets/js/components/MainContent/BrowseMode/Table/Table.vue]: echos and console' logs should probably be remove

**Review by James-N-M — APPROVED** [review]

File: resources/assets/js/components/Pages/PrequelError.vue [hunk 6]
```diff
@@ -72,7 +75,7 @@
         }
 
         if (suggestionCollection.length === 0) {
-          suggestionCollection.push(trans('error_page.no_suggestions'));
+          suggestionCollection.push('Prequel could not suggest any fixes.');
```
Comment (QuintenJustus) [inline resources/assets/js/components/Pages/PrequelError.vue]: `suggestionCollection.push(trans('error_page.no_suggestions'));`	 ?

Comment (M4ST3RX) [reply resources/assets/js/components/Pages/PrequelError.vue]: Function trans is not found when it is inside a function for me

Comment (QuintenJustus) [reply resources/assets/js/components/Pages/PrequelError.vue]: Besides this, I think it's all set for merge! Nice work :+1:

Comment (QuintenJustus) [reply resources/assets/js/components/Pages/PrequelError.vue]: Hm, I'll fix it then

**Pull request merged**